### ISSUE TITLE
Compile error for overlapping nets with Tf.Keras metrics

### ISSUE THREAD

**Issue by bolau** [tensorflow/tensorflow#26927]
**System information**
- Have I written custom code: yes
- OS Platform and Distribution: macOS 10.14
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below):  broken with 1.13.1, works with 1.12.0
- Python version: 3.6.3

**Describe the current behavior**
Compiling concatenated networks with Keras metrics causes an InvalidArgumentError in the input of the second network (You must feed a value for placeholder tensor 'dense_5_target' with dtype float and shape [?,?]). This used to work in 1.12.0 but broke with 1.13.1. Without metrics it's not an issue.

**Describe the expected behavior**
The networks should be trainable individually and in the concatenated version, no matter if we specify metrics or not.

**Code to reproduce the issue**
```
# coding: utf-8

import numpy as np
import tensorflow as tf
from tensorflow.keras.models import Model, Sequential
from tensorflow.keras.layers import Dense, Input

# works with 1.12, fails with 1.13.1
print(tf.__version__)

# Layer sizes of net 1 and 2
n_input_1 = 5
n_output_1 = 4

n_input_2 = 4
n_output_2 = 2

# Generating dummy data
N = 13
x1 = np.random.rand(N, n_input_1)
y1 = np.random.rand(N, n_output_1)

x2 = np.random.rand(N, n_output_1)
y2 = np.random.rand(N, n_output_2)


# Build net1, net2 and net_full
# net_full concats net1 and net2
def build_models():
    input_layer_1 = Input(shape=(n_input_1,))
    output_layer_1 = Dense(n_output_1)(input_layer_1)
    net1 = Model(inputs=input_layer_1, outputs=output_layer_1, name="net1")

    input_layer_2 = Input(shape=(n_input_2,), name = "topmodel_input")
    output_layer_2 = Dense(n_output_2)(input_layer_2)
    net2 = Model(inputs=input_layer_2, outputs=output_layer_2, name="net2")

    net_full = Model(inputs=input_layer_1, outputs=net2(net1.output))
    
    return net2, net_full

net2, net_full = build_models()
# compile with passing metrics only to net2 --> runs
net2.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
net_full.compile(optimizer='rmsprop', loss='categorical_crossentropy')
net_full.fit(x1, y2)

net2, net_full = build_models()
# compile with passing metrics only to net_full --> runs
net2.compile(optimizer='rmsprop', loss='categorical_crossentropy')
net_full.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
net_full.fit(x1, y2)

net2, net_full = build_models()
# compile with passing metrics to both net2 AND net_full --> crashes
net2.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
net_full.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
net_full.fit(x1, y2)
```

I think this is a bug. If you consider this improper use instead, please let me know.

Best, Boris

**Comment by foxik** [comment]
For TF 2.0, it has been reported as https://github.com/tensorflow/tensorflow/issues/26738. From what you say, it seems that it also affects 1.13.

**Comment by pavithrasv** [comment]
659c981 should fix this issue. Please give it a try in the next nightly and let me know if it works as expected.

Thank you!

**Comment by pavithrasv** [comment]
The fix was verified by @foxik in #26738 . Thank you!
